Introduce support for the "jdbc-uri" property in k8s service binding

Sgitario opened the pull request:
Plus, log a warning when the uri property does not have the expected format Fix https://github.com/quarkusio/quarkus/issues/30682

RuanNunes commented on extensions/kubernetes-service-binding/runtime/src/main/java/io/quarkus/kubernetes/service/binding/runtime/DatasourceServiceBindingConfigSourceFactory.java:
```diff
@@ -79,23 +79,38 @@ protected String formatUrl(String urlFormat, String type, String host, String da
 
+    private boolean configureUrlPropertyUsingKey(Map<String, String> properties, String key) {
+        String value = serviceBinding.getProperties().get(key);
+        if (value != null) {
+            if (value.startsWith(prefix)) {
+                properties.put(urlPropertyName, value);
+                return true;
+            } else {
+                log.warnf("The value '%s' from the property '%s' does not start with '%s'. It will be ignored.",
+                        value, key, prefix);
+            }
+        }
+
+        return false;
```
```suggestion
       if (value == null) {
            return false;
        }
        if (value.startsWith(prefix)) {
            properties.put(urlPropertyName, value);
            return true;
        }
        log.warnf("The value '%s' from the property '%s' does not start with '%s'. It will be ignored.",
        value, key, prefix);
```

RuanNunes replied:
suggestion to simplify the logic

Sgitario replied:
Updated

Sgitario committed:
Introduce support for the "jdbc-uri" property in k8s service binding

Plus, log a warning when the uri property does not have the expected format
Fix https://github.com/quarkusio/quarkus/issues/30682
